using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace exception_Handing_ex
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int[] num = { 1, 2, 3 };
            try
            {
                Console.WriteLine(num[1]);
                Console.WriteLine(num[3]);//error will come
            }
            catch (Exception ee) // code will execute only if run time error occurs
            {
                Console.WriteLine("catch block is executing");
                Console.WriteLine(ee.ToString() );
            }
            finally //code will execute whehter error occurs or not
            {
                Console.WriteLine("finally block is executing");
               
            }

            Console.WriteLine("bye");

            Console.ReadLine();
            
        }
    }
}
